### PR TITLE
Add a withIndent CodeBlock builder function.

### PR DESCRIPTION
**Pull request by PaulWoitaschek**
As suggested in this issue https://github.com/square/kotlinpoet/issues/1072 I've implemented the `withIndent` helper function.

### CODE REVIEW

File: kotlinpoet/src/main/java/com/squareup/kotlinpoet/CodeBlock.kt [hunk 1]
```diff
@@ -492,3 +492,12 @@ public fun Collection<CodeBlock>.joinToCode(
 public inline fun buildCodeBlock(builderAction: CodeBlock.Builder.() -> Unit): CodeBlock {
   return CodeBlock.builder().apply(builderAction).build()
 }
+
+/**
+ * Calls [CodeBlock.Builder.indent] then executes the the provided [builderAction] on the
```
Comment (Egorand) [inline kotlinpoet/src/main/java/com/squareup/kotlinpoet/CodeBlock.kt:497]: nit:
```suggestion
 * Calls [CodeBlock.Builder.indent] then executes the provided [builderAction] on the
```

**Review by Egorand — APPROVED** [review]
Awesome, I like this!

File: kotlinpoet/src/main/java/com/squareup/kotlinpoet/CodeBlock.kt (hunk 1 above)
Comment (PaulWoitaschek) [reply kotlinpoet/src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: You're [a genius](https://www.reddit.com/r/explainlikeimfive/comments/3ze4ca/eli5_why_does_the_the_human_mind_ignore_the/) :joy: . Fixed!

Comment (Egorand) [reply kotlinpoet/src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: Officially a 10x engineer!

**Comment by Egorand** [comment]
@PaulWoitaschek please sign our [CLA](https://spreadsheets.google.com/spreadsheet/viewform?formkey=dDViT2xzUHAwRkI3X3k5Z0lQM091OGc6MQ&ndplr=1) and I'm ready to merge. Thanks!

**Comment by PaulWoitaschek** [comment]
Signed!
